test/libyul/objectCompiler/eof/rjumps.yul

```diff
@@ -0,0 +1,40 @@
+object "a" {
+    code {
+        if true {
+            mstore(0, 1)
+        }
+
+        return(0, 32)
+    }
+}
+
+// ====
+// EVMVersion: >=prague
+// bytecodeFormat: >=EOFv1
+// ----
+// Assembly:
+//     /* "source":49:53   */
+//   0x01
+//     /* "source":46:70   */
+//   rjumpi{tag_1}
+//     /* "source":22:112   */
+// tag_2:
```

Comment:
@rodiazet I just noticed an odd thing while reviewing #15550. `RJUMPI` seems to always insert an extra tag that often goes unused. Somehow it does get used in the case you added here but you can see it in `objectCompiler/eof/functions.yul` in the `CALLF` PR. It's also easily reproducible without functions:

`test.yul`:
```yul
{
    if calldataload(0) {
        return(0, 0)
    }
}
```
```bash
solc --strict-assembly test.yul --asm --debug-info none --experimental-eof-version 1 --evm-version prague
```
```
  0x00
  calldataload
  rjumpi{tag_1}
tag_2:
  stop
tag_1:
  0x00
  dup1
  return
```

It does not happen with `JUMPI` so it must be something introduced by this PR:
```bash
solc --strict-assembly test.yul --asm --debug-info none --evm-version prague
```
```
  0x00
  calldataload
  tag_1
  jumpi
  stop
tag_1:
  0x00
  dup1
  return
```

Reply:
It's related to lack of optimizations for EOF. In this case `PeepholeOptimizer` is disabled for EOF. For legacy `tag_2` is removed by the `UnreachableCode` optimizer step. I checked that for legacy the `tag_2` is present in the code section `0` before optimization.

Reply:
Ah, ok then. I didn't consider that it may be the optimizer's responsibility but it makes sense.